packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -436,32 +436,49 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | TeamsCa
   );
   constructor(
     callClient: StatefulCallClient,
-    locatorOrTargetCalless: CallAdapterLocator | StartCallIdentifier[],
+    call: Call,
+    callAgent: AgentType,
+    deviceManager: StatefulDeviceManager,
+    options?: AzureCommunicationCallAdapterOptions | TeamsAdapterOptions
+  );
+  constructor(
+    callClient: StatefulCallClient,
+    overloadedParam: CallAdapterLocator | StartCallIdentifier[] | Call,
     callAgent: AgentType,
     deviceManager: StatefulDeviceManager,
     options?: AzureCommunicationCallAdapterOptions | TeamsAdapterOptions
   ) {
     this.bindPublicMethods();
     this.callClient = callClient;
     this.callAgent = callAgent;
-    this.targetCallees =
-      getLocatorOrTargetCallees(locatorOrTargetCalless) === true
-        ? (locatorOrTargetCalless as StartCallIdentifier[])
-        : undefined;
-    this.locator =
-      getLocatorOrTargetCallees(locatorOrTargetCalless) === false
-        ? (locatorOrTargetCalless as CallAdapterLocator)
-        : undefined;
+
+    let overloadedParamAsCall: Call | undefined = undefined;
+    if (isTargetCallees(overloadedParam)) {
+      this.targetCallees = overloadedParam;
+    } else if (isCall(overloadedParam)) {
+      overloadedParamAsCall = overloadedParam;
+    } else {
+      this.locator = overloadedParam;
+    }
+
     this.deviceManager = deviceManager;
-    const isTeamsMeeting = this.locator ? 'meetingLink' in this.locator || 'meetingId' in this.locator : false;
+
+    const isTeamsMeeting = this.locator
+      ? 'meetingLink' in this.locator || 'meetingId' in this.locator
+      : !!overloadedParamAsCall?.info.threadId;
```

Comment:
Do we know that we get the thread id in this case if the call is not in the connected state? if not do we need to do anything here so we can still make the thread for the cwc adapter? Just thinking that this is technically possible to use when the call is in a early state like connecting or EarlyMedia

Reply:
Hmm dang you're right, connecting and lobby both don't have the threadId yet, only connected does
![image](https://github.com/user-attachments/assets/a17f97df-3f7b-4281-bebe-37f6b6f2b4c6)

Reply:
Updated to have the isTeamsMeeting be calculated on onClientStateChange